Feature/hhausner a5818 compatability

hausnerh opened the pull request:
### Description

Add compatibility for ICARUS boardreader code to use A5818 cards

### Testing details

TBD

SilentSegfault committed:
Merge pull request #169 from SBNSoftware/release/v1_10_03

Release/v1 10 03

hausnerh committed:
Update PhysCrateData_generator.cc

Edit ReadFirmwareVersion to handle A5818 cards. Generator now looks for a "AX818" fhicl parameter, which defaults to 3. If set to 5 it looks for an A5818 card.

hausnerh committed:
Update A2795Board.cc

Removed depreciated use of CAENComm_OpenDevice
Replaced with CAENComm_OpenDevice2.
The difference is in the second argument being a pointer to the link number as opposed to the raw link number.

mvicenzi commented:
Hi @hausnerh, would you be able to add the switcheroo between `cvA3818` and `cvA5818` also for `CAENV1730Readout::GetSWInfo()`? 

I believe that's the only other place where we explicitly refer to an A3818 connection (the optical link is opened using `CAENDGTZ::OpenDigitizer()` which should handle things internally).

hausnerh committed:
Update CAENConfiguration.hh to add AX818 configuration

add parameter to set if we are using the A5818 cards or not

hausnerh committed:
Update CAENConfiguration.cc for A5818 compatibility

Add parameter for configuring running with A5818 cards. Default to A3818 (our standard running configuration)

hausnerh committed:
Update CAENConfiguration.cc

This is Common, not SBND specific. ICARUS also uses this configuration class.

hausnerh committed:
Update CAENV1730Readout_generator.cc for A5818 compatibility

use configuration in fCAEN to toggle which PCIe board is being used

hausnerh commented:
> Hi @hausnerh, would you be able to add the switcheroo between `cvA3818` and `cvA5818` also for `CAENV1730Readout::GetSWInfo()`?
> 
> I believe that's the only other place where we explicitly refer to an A3818 connection (the optical link is opened using `CAENDGTZ::OpenDigitizer()` which should handle things internally).

Added

dtorretta56 reviewed (approved):
OK to me. 
thanks!

SilentSegfault committed:
addressed compilation errors
